src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_MULTIPLIER = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+    private final int capacity;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size / LOAD_FACTOR > capacity) {
+            resize();
+        }
+        Node<K, V> oldNode = table[getIndexHash(key)];
+        while (oldNode != null) {
+            if (key == null && oldNode.key == null || key != null && key.equals(oldNode.key)) {
+                size--;
+            }
+            oldNode = oldNode.next;
+        }
+        Node<K, V> newNode = new Node<>(getIndexHash(key), key, value, table[getIndexHash(key)]);
+        table[getIndexHash(key)] = newNode;
```

Comment:
1. using this approach you can have a few nodes with the same key - it is not correct
2. do not call getIndex more than 1 time